src/Chapter2.hs

```diff
@@ -727,7 +739,7 @@ value of the element itself
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
+smartReplicate l = concat (map (\x -> replicate x x) l)
```

Comment:
This is correct!

There is also a standard function `concatMap` which unite together `concat` and `map`. So this could be written like: ` concatMap (\x -> ...) l`. Btw, with such implementation, you can try to eta reduce it 🙂

Reply:
Wow! This seems like `flatMap` from other languages.